app/service/file_svc.py

```diff
@@ -78,15 +89,32 @@ def __init__(self, plugins, exfil_dir):
         :param name:
         :return: a tuple (file_path, contents)
         """
-        _, file_name = await self.find_file_path(name, location='payloads')
-        if file_name:
-            with open(file_name, 'rb') as file_stream:
-                return name, file_stream.read()
-        _, file_name = await self.find_file_path('%s.xored' % (name,), location='payloads')
-        if file_name:
-            return name, xor_file(file_name)
+        for loc in ['payloads', 'data']:
+            _, file_name = await self.find_file_path(name, location=loc)
+            if file_name:
+                with open(file_name, 'rb') as file_stream:
+                    return name, file_stream.read()
+            _, file_name = await self.find_file_path('%s.xored' % (name,), location=loc)
+            if file_name:
+                return name, xor_file(file_name)
         raise FileNotFoundError
 
+    async def save_file(self, request):
+        """
+        Save a (payload) file to the stockpile
+        :param name: filename
+        :param content: content to save
+        :param xored: whether or not to xor the contents when saving
+        :return: full path of the saved file
+        """
+        filebase = 'data/payloads/'
```

Comment:
what uses this function? it says in the doc string that it saves to stockpile..?

Reply:
This is used in the chain API to handle file uploads from the UI.